csxtools/utils.py

```diff
@@ -275,8 +275,9 @@ def calculate_flatfield(image, limits=(0.6, 1.4)):
     return flat
 
 
-def get_fastccd_flatfield(light, dark, flat=None, limits=(0.6, 1.4)):
-    """Calculate a flatfield from two headers
+
+def get_fastccd_flatfield(light, dark, flat=None, limits=(0.6, 1.4), half_mode=False, half_args =(7, 486)):
```

Comment:
Where possible, we try to avoid "coupled optional arguments": situations where changing the _value_ of one parameter changes the _interpretation or applicability_ of another. This scenario is ripe for confusion or frustration:

```py
get_fastccd_flatfield(light, dark, half_args=(6, 486))
```

Here, I'm trying to tune the `half_args` and I might wonder why my changes aren't being used. (It's because `half_mode` is still `False`.)

There's no _great_ solution here, but I might suggest using just one parameter. Maybe call it `half_interval`.

```py
half_interval=False  # default, do nothing
half_interval=True  # use (7, 486) hard-coded in function
half_interval=(..., ...)  # use custom interval
```

In this way, `True` means "Yes do it and I accept whatever the default intervals is, which may change over time if you re-tune this."

Reply:
good practical advice for improved user experience.  will make the change.